src/Hamiltonians/MatrixHamiltonian.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    MatrixHamiltonian(mat::AbstractMatrix{T}; starting_address=1) <: AbstractHamiltonian{T}
+Wrap an abstract matrix `mat` as an `AbstractHamiltonian` object for use with
+regular `Vector`s indexed by integers. Works with stochatic methods of
+[`lomc!()`](@ref). Optionally, a `starting_address` can be provided.
+
+Specialised methods are implemented for sparse matrices of type `AbstractSparseMatrixCSC`.
+"""
+struct MatrixHamiltonian{T,AM,H} <: AbstractHamiltonian{T}
+    m::AM
+    starting_index::Int
+end
+
+function MatrixHamiltonian(m::AM; starting_address=axes(m, 2)[1]) where AM <:AbstractMatrix
+    s = length.(axes(m))
+    @assert s[1]==s[2] "Matrix needs to be square, got $s."
+    i = Int(starting_address)
+    @assert minimum(axes(m, 2)) ≤ i ≤ maximum(axes(m, 2)) "Invalid index $starting_address."
+    MatrixHamiltonian{eltype(m), AM, ishermitian(m)}(m, i)
+end
+
+LOStructure(::Type{<:MatrixHamiltonian{<:Any,<:Any,true}}) = Hermitian()
+LOStructure(::Type{<:MatrixHamiltonian}) = AdjointKnown()
+LinearAlgebra.adjoint(mh::MatrixHamiltonian) = MatrixHamiltonian(mh.m')
+LinearAlgebra.adjoint(mh::MatrixHamiltonian{<:Any,<:Any,true}) = mh
+
+
+starting_address(mh::MatrixHamiltonian) = mh.starting_index
+
+dimension(::Type{T}, mh::MatrixHamiltonian) where T = T(length(axes(mh.m,2)))
```

Comment:
Perhaps `length(axes(...))` should be replaced by `size(...)`

Reply:
Makes sense. Thanks!